python_modules/libraries/dagster-k8s/dagster_k8s/launcher.py

```diff
@@ -445,8 +445,10 @@ def check_run_worker_health(self, run: DagsterRun):
                 WorkerStatus.FAILED, "Run has not completed but K8s job has no active pods"
             )
 
-        if status.failed:
-            return CheckRunHealthResult(WorkerStatus.FAILED, "K8s job failed")
         if status.succeeded:
             return CheckRunHealthResult(WorkerStatus.SUCCESS)
+        else:
+            if status.failed > 0 and not status.active:
+                return CheckRunHealthResult(WorkerStatus.FAILED, "K8s job failed")
+
```

Comment:
```suggestion
        if status.failed and not status.active:
            return CheckRunHealthResult(WorkerStatus.FAILED, "K8s job failed")
```
This seems a bit more minimal, and I think it should be the same thing.

Reply:
What about the case where there is a >1 backoff limit, one run fails, and then the next succeeds? My assumption is that it would have:
- active: 0
- failed: 1
- succeeded: 1

I'm certainly not as familiar with this part of the k8s APIs, so defer to others, but just want to make sure that use case works as expected and doesn't declare a run failure.

Reply:
@adam-bloom: in that case you listed, wouldn't the "if status.succeeded:" case above still return? (with both your code and with @deepyaman 's suggestion)

Reply:
ah yeah, duh 🤦 . As long as we flip the order from the original implementation, that should cover it.